tests/components/devialet/test_media_player.py

```diff
@@ -105,130 +117,139 @@ async def test_media_player_playing(
     hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
 ) -> None:
     """Test the Devialet configuration entry loading and unloading."""
-    await async_setup_component(hass, "homeassistant", {})
-    entry = await setup_integration(hass, aioclient_mock)
 
-    assert entry.entry_id in hass.data[DOMAIN]
-    assert entry.state is ConfigEntryState.LOADED
+    with patch("devialet.DevialetApi.upnp_available", return_value=True):
+        await async_setup_component(hass, "homeassistant", {})
+        entry = await setup_integration(hass, aioclient_mock)
 
-    await hass.services.async_call(
-        "homeassistant",
-        SERVICE_UPDATE_ENTITY,
-        {ATTR_ENTITY_ID: [f"{MP_DOMAIN}.{NAME.lower()}"]},
-        blocking=True,
-    )
+        assert entry.entry_id in hass.data[DOMAIN]
+        assert entry.state is ConfigEntryState.LOADED
 
-    state = hass.states.get(f"{MP_DOMAIN}.{NAME.lower()}")
-    assert state.state == MediaPlayerState.PLAYING
-    assert state.name == NAME
-    assert state.attributes[ATTR_MEDIA_VOLUME_LEVEL] == 0.2
-    assert state.attributes[ATTR_MEDIA_VOLUME_MUTED] is False
-    assert state.attributes[ATTR_INPUT_SOURCE_LIST] is not None
-    assert state.attributes[ATTR_SOUND_MODE_LIST] is not None
-    assert state.attributes[ATTR_MEDIA_ARTIST] == "The Beatles"
-    assert state.attributes[ATTR_MEDIA_ALBUM_NAME] == "1 (Remastered)"
-    assert state.attributes[ATTR_MEDIA_TITLE] == "Hey Jude - Remastered 2015"
-    assert state.attributes[ATTR_ENTITY_PICTURE] is not None
-    assert state.attributes[ATTR_MEDIA_DURATION] == 425653
-    assert state.attributes[ATTR_MEDIA_POSITION] == 123102
-    assert state.attributes[ATTR_MEDIA_POSITION_UPDATED_AT] is not None
-    assert state.attributes[ATTR_SUPPORTED_FEATURES] is not None
-    assert state.attributes[ATTR_INPUT_SOURCE] is not None
-    assert state.attributes[ATTR_SOUND_MODE] is not None
-
-    with patch(
-        "homeassistant.components.devialet.DevialetApi.playing_state",
-        new_callable=PropertyMock,
-    ) as mock:
-        mock.return_value = MediaPlayerState.PAUSED
-
-        await hass.config_entries.async_reload(entry.entry_id)
-        await hass.async_block_till_done()
-        assert (
-            hass.states.get(f"{MP_DOMAIN}.{NAME.lower()}").state
-            == MediaPlayerState.PAUSED
+        await hass.services.async_call(
+            "homeassistant",
+            SERVICE_UPDATE_ENTITY,
+            {ATTR_ENTITY_ID: [f"{MP_DOMAIN}.{NAME.lower()}"]},
+            blocking=True,
```

Comment:
Why do we do this? Why don't we let the integration update the integration naturally?

Reply:
Without the forced update, it shows: `KeyError: 'volume_level'`. How is the update triggered in the tests?

Reply:
We either freeze time or we the library mock pushes data. Is this a push or pull based integration?

Reply:
It's a pull based integration.

Reply:
No, by skipping time using the freezer.

If its push based we should use the mock of the library we have to push an update. Check knocki for an example

Reply:
I see the freezegun used like `@pytest.mark.freeze_time("2025-01-01T12:00:00Z")` but how does this work for the HA update? I think I don't get it. When I freeze time, setup integration and skip time, the update did not occur.

Reply:
But you just said it's a push based integration, so you can't trigger natural updates by skipping time. (You do that with freezer.tick btw)

Reply:
> It's a pull based integration. You mean freeze time as in: `await asyncio.sleep(6)`?

Did I? It's `local_polling`, to be more accurate.

Reply:
LOL, I am not sure how I misread that. Sorry.

You need to use `freezer.tick(..)` to skip time

Reply:
Thanks! That did the t(r)ick